sema4ai/vscode-client/src/extension.ts

```diff
@@ -1003,3 +1018,69 @@ export async function getLanguageServerPythonInfo(): Promise<InterpreterInfo | u
 export const collapseAllEntries = async (): Promise<void> => {
     vscode.commands.executeCommand(`workbench.actions.treeView.${TREE_VIEW_SEMA4AI_TASK_PACKAGES_TREE}.collapseAll`);
 };
+
+export async function dropDataSource(entry?: RobotEntry) {
+    if (!(await verifyDataExtensionIsInstalled())) {
+        return;
+    }
+    const dataServerStatus = await commands.executeCommand(DATA_SERVER_STATUS_COMMAND_ID);
+    if (!dataServerStatus["success"]) {
+        window.showErrorMessage("Unable to get the data server status.");
+        return;
+    }
+
+    let datasource: DatasourceInfo | undefined;
+    if (!entry) {
+        let selectedActionPackage = await findActionPackagePath({ includeSemaOrg: false });
+        if (!selectedActionPackage) {
+            return;
+        }
+        let listDataSourcesResult = await vscode.commands.executeCommand(SEMA4AI_LIST_ACTIONS_INTERNAL, {
+            "action_package": selectedActionPackage.toString(),
+            "collect_datasources": true,
+        });
+
+        if (!listDataSourcesResult["success"]) {
+            window.showErrorMessage(listDataSourcesResult["message"] || "Unknown error");
+        }
+
+        const dataSources: DatasourceInfo[] = listDataSourcesResult["result"].filter((ds) => {
+            return ds.kind === "datasource";
+        });
+
+        if (dataSources.length === 0) {
+            window.showInformationMessage("No Data Sources found.");
+            return;
+        }
+
+        let captions: QuickPickItemWithAction[] = new Array();
+        for (const source of dataSources) {
+            let caption: QuickPickItemWithAction = {
+                "label": getDataSourceCaption(source),
+                "action": source,
+            };
+            captions.push(caption);
+        }
+        let selectedItem = await showSelectOneQuickPick(captions, "Please select which Data Source to drop.");
+
+        if (!selectedItem) {
+            return;
+        }
+
+        datasource = selectedItem.action;
+    } else {
+        datasource = entry?.extraData?.datasource;
+    }
+
+    const result = await langServer.sendRequest("dropDataSource", {
+        "datasource": datasource,
+        "data_server_info": dataServerStatus["data"],
+    });
+
+    if (!result["success"]) {
+        window.showErrorMessage(result["message"] || "Unknown error");
+        return;
+    }
+
+    window.showInformationMessage("Data Source dropped successfully.");
```

Comment:
I think that in the success case, the `result['message']` should be shown (so, we can say that either it was dropped successfully or it wasn't dropped because it didn't exist -- that's not an error, so, it's a 'success' with a 'message').